### DISCUSSION TITLE
'Submission' object has no attribute 'remote_root'

### DISCUSSION THREAD

**Discussion by maverick422** [deepmodeling/dpdispatcher#175] [Q&A] [answered]
### Report error message:

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/public/home/huangan21/dpgen/bin/dpgen", line 10, in <module>
    sys.exit(main())
  File "/public/home/huangan21/dpgen/lib/python3.8/site-packages/dpgen/main.py", line 175, in main
    args.func(args)
  File "/public/home/huangan21/dpgen/lib/python3.8/site-packages/dpgen/generator/run.py", line 2944, in gen_run
    run_iter (args.PARAM, args.MACHINE)
  File "/public/home/huangan21/dpgen/lib/python3.8/site-packages/dpgen/generator/run.py", line 2920, in run_iter
    run_model_devi (ii, jdata, mdata)
  File "/public/home/huangan21/dpgen/lib/python3.8/site-packages/dpgen/generator/run.py", line 1297, in run_model_devi
    submission.run_submission()
  File "/public/home/huangan21/dpgen/lib/python3.8/site-packages/dpdispatcher/submission.py", line 182, in run_submission
    self.handle_unexpected_submission_state()
  File "/public/home/huangan21/dpgen/lib/python3.8/site-packages/dpdispatcher/submission.py", line 219, in handle_unexpected_submission_state
    f"Meet errors will handle unexpected submission state.\n"
AttributeError: 'Submission' object has no attribute 'remote_root'

### machine.json file:

{
  "api_version": "1.0",
  "train": [
    {
      "command": "dp2",
      "machine": {
          "context_type":"local",
        "batch_type": "PBS",
        "machine_type": "PBS",
        "local_root":"./",
        "remote_root":"/public/home/huangan21/BPhosphorus/dpgen/run/main/dp",
	"hostname" :"node008",
	"_port" :22,
	"_username":"huangan21",
	"_work_path" : "/public/home/huangan21/BPhosphorus/dpgen/run"
      },
      "resources": {
        "number_node": 1,
        "cpu_per_node": 2,
        "gpu_per_node": 2,
        "queue_name": "gpu",
        "group_size": 2,
        "para_deg": 3,
        "_source_list": [  ]
      }
    }
  ],
  "model_devi":[
    {
      "command": "lmp2",
      "machine":{
          "context_type":"local",
        "batch_type": "PBS",
        "machine_type": "PBS",
        "local_root":"./",                                                       
        "remote_root":"/public/home/huangan21/BPhosphorus/dpgen/run/main/lmp",
          "hostname" :	"node008",
	"_port" :22,
	"_username":"huangan21",
	"_work_path" : "/public/home/huangan21/BPhosphorus/dpgen/run"
      },
      "resources": {
        "number_node": 1,
        "cpu_per_node": 2,
        "gpu_per_node": 2,
        "queue_name": "gpu",
        "group_size": 2,
        "_source_list": [ ]
      }
    }
  ],
  "fp":[
    {
      "command": "mpirun -np 12  vasp610",
      "machine":{
          "context_type":"local",
        "batch_type": "PBS",
        "machine_type": "PBS",
        "local_root":"./", 
        "remote_root":"/public/home/huangan21/BPhosphorus/dpgen/run/main/fp", 
	"hostname" :"node008",
	"_port" :22,
	"_username":"huangan21",
	"_work_path" : "/public/home/huangan21/BPhosphorus/dpgen/run"
      },
      "resources": {
        "number_node": 1,
        "cpu_per_node": 12,
        "gpu_per_node": 0,
        "queue_name": "fast",
        "group_size": 10,
        "_source_list": [  ]
      }
    }
  ]
}

thanks a lot!

**Comment by njzjz** [comment, marked as answer]
Please upgrade to the latest version.

**Reply by maverick422** [reply]
thanks a lot！

**Comment by Kangxuxin** [comment]
After 'pip install dpdispatcher -U'

WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f8cd7065840>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/dpdispatcher/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f8cd7065b70>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/dpdispatcher/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f8cd7065e10>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/dpdispatcher/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f8cd7065fc0>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/dpdispatcher/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f8cd7066170>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/dpdispatcher/
ERROR: Could not find a version that satisfies the requirement dpdispatcher (from versions: none)
ERROR: No matching distribution found for dpdispatcher
WARNING: There was an error checking the latest version of pip.

**Comment by Kangxuxin** [comment]
Then 'pip install --upgrade pip'

Report error message:
Requirement already satisfied: pip in /public/home/duanxiangmei/softwore/deepmd-kit/lib/python3.10/site-packages (22.1.2)
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f5a7cc81810>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/pip/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f5a7cc81b40>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/pip/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f5a7cc81cf0>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/pip/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f5a7cc81ea0>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/pip/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f5a7cc82050>: Failed to establish a new connection: [Errno 101] Network is unreachable')': /simple/pip/
WARNING: There was an error checking the latest version of pip.

Thanks a lot
